core/src/main/java/com/graphhopper/util/shapes/Polygon.java

```diff
@@ -92,6 +96,11 @@ public boolean contains(GHPoint point) {
         return contains(point.lat, point.lon);
     }
 
+    @Override
+    public boolean intersect(Shape o) {
+        return false;
```

Comment:
Here it should throw an UnsupportedException and not return a potentially wrong result.